Fdd - blog post

upsd opened the pull request:
A post on Fear Driven Development.

I still need imagery, and will hopefully source this today.

Commit:
A first stab at template of article.

Commit:
Add links to be utilised in section.

Commit:
Start of intro.

Commit:
Add some more info to first section.

Commit:
Add some examples of what can cause fear.

Commit:
Reword wording of section.

Commit:
Subsection added to describe causes of fear.

Commit:
More info of unfamiliar with codebase.

Commit:
Cloning the project.

Commit:
Summary sentence.

Commit:
New to the team.

Commit:
Lack of context.

Commit:
Amend context section.

Commit:
Pressure.

Commit:
Where the fear leads (title).

Commit:
Where the fear leads (intro).

Commit:
Append-only programming.

Commit:
Copy and paste programming.

Commit:
Tunnel vision programming.

Commit:
Where the fear stops (intro).

Commit:
Tests.

Commit:
Taking a step back (title).

Commit:
Taking a step back (body).

Commit:
Closing section.

Commit:
Formatting.

Commit:
Add tags and number titles.

WencesLlobet commented on src/_posts/2020-01-14-fear-driven-development.md:
````diff
@@ -0,0 +1,318 @@
+}
+```
+
+Fortunately your test might catch that extra `0` you added and you can revert your change. No fear.
````
I would comment as well that tests are a great way to clarify context across team mates "is that what we want?" this benefit helped me a lot in my last client.

upsd replied:
This is a good point, I will take a look 👍 

Thank you for taking the time to suggest this.

WencesLlobet reviewed (commented):
It looks nice Sam. I added a small comment to contribute somehow :)

Commit:
Add image.

The image does not look too great on larger viewports, maybe we can add some inline CSS, or people can stare at my forehead instead.

Commit:
Add Lau's suggestion on testing.

neilkidd-codurance commented on src/_posts/2020-01-14-fear-driven-development.md:
```diff
@@ -0,0 +1,320 @@
+assuming that you incorrectly framed a piece of information, the recipient(s) of this will now have
+a broken-context.
+
+Many of have a broken-context, and some examples are below:
```
This sentence needs checking - Is there something missing? EG: 
"Many of  **the examples** have a broken-context, and some examples are below:"

neilkidd-codurance commented on src/_posts/2020-01-14-fear-driven-development.md:
```diff
@@ -0,0 +1,320 @@
+* unsure why a particular story is relevant
+* uncertain how your team fits into the bigger picture
+
+To keep scope-creep from happening here, I will leave it as that. Just be aware that a lack of
```
Should this read: "I will leave it **at** that." ?

neilkidd-codurance commented on src/_posts/2020-01-14-fear-driven-development.md:
```diff
@@ -0,0 +1,320 @@
+story of pressure, and how it can evoke fear.
+
+The pressure applied to you by your manager might make you fearful of what you are doing. It might
+even result in you doing something that is rushed an could actually make the system worse.
```
an => and?

haletothewood commented on src/_posts/2020-01-14-fear-driven-development.md:
```diff
@@ -0,0 +1,318 @@
+- decision-making
+- legacy-code
+---
+On the internet there are a few different perspectives on Fear Driven Development (FDD). The
```
```suggestion
On the internet, there are a few different perspectives on Fear Driven Development (FDD). The
```

haletothewood commented on src/_posts/2020-01-14-fear-driven-development.md:
```diff
@@ -0,0 +1,318 @@
+
+Close your eyes and imagine someone looking at some code on a computer screen, they need to make
+a change to that code. They then do something you would not expect, and you want to ask them why
+they chose to make that decision (you can't, because they are in your imagination). This decision
```
```suggestion
they chose to make that decision (you can't because they are in your imagination). This decision
```

haletothewood commented on src/_posts/2020-01-14-fear-driven-development.md:
```diff
@@ -0,0 +1,318 @@
+Nice and simple right? That definition is simple, and deliberately so. The focus of this post is not
+to define what legacy code is.
+
+In another post by [Scott Hanselman](https://www.hanselman.com/blog/FearDrivenDevelopmentFDD.aspx)
```
```suggestion
In another post by [Scott Hanselman](https://www.hanselman.com/blog/FearDrivenDevelopmentFDD.aspx),
```

haletothewood commented on src/_posts/2020-01-14-fear-driven-development.md:
```diff
@@ -0,0 +1,318 @@
+### New to the team
+Day one, an exciting day indeed. It is your first day on your new team, you want to be accepted by
+your new team, and so feel the need to prove yourself (no bread jokes here). It is the standup and
+you are extremely excited to pick up a story, so like Mr Unfamiliar in the previous scenario, you
```
```suggestion
you are extremely excited to pick up a story, so like Mr. Unfamiliar in the previous scenario, you
```

haletothewood commented on src/_posts/2020-01-14-fear-driven-development.md:
```diff
@@ -0,0 +1,318 @@
+
+The scenario I have just described can create fear.
+### Lack of context
+This can be found in the both the previous sections, in that Mr Unfamiliar:
```
```suggestion
This can be found in the both the previous sections, in that Mr. Unfamiliar:
```

haletothewood replied:
```suggestion
This can be found in both the previous sections, in that Mr Unfamiliar:
```

haletothewood commented on src/_posts/2020-01-14-fear-driven-development.md:
```diff
@@ -0,0 +1,318 @@
+* did not have any documentation to work with
+* had not previously worked with the codebase
+
+and Mrs New:
```
```suggestion
and Mrs. New:
```

neilkidd-codurance commented on src/_posts/2020-01-14-fear-driven-development.md:
```diff
@@ -0,0 +1,320 @@
+## A final message
+This post provides examples of how fear can start, and where it can lead. As a reader (and maybe a
+developer/programmer/wizard) you are in control, it is your decisions and those of your teammates
+that will decide whether your code is consumed by fear, or guarded by courage.
```
Maybe end this sentence with something like:
", or supported with confidence and courage."

haletothewood commented on src/_posts/2020-01-14-fear-driven-development.md:
```diff
@@ -0,0 +1,318 @@
+
+* possibly did not have sufficient knowledge of the system in her new team
+
+Therefore, these can categorised as symptoms of fear, they culminate with other factors to create
```
```suggestion
Therefore, these can be categorised as symptoms of fear, they culminate with other factors to create
```

haletothewood commented on src/_posts/2020-01-14-fear-driven-development.md:
```diff
@@ -0,0 +1,318 @@
+story of pressure, and how it can evoke fear.
+
+The pressure applied to you by your manager might make you fearful of what you are doing. It might
+even result in you doing something that is rushed an could actually make the system worse.
```
```suggestion
even result in you doing something that is rushed and could actually make the system worse.
```

haletothewood commented on src/_posts/2020-01-14-fear-driven-development.md:
````diff
@@ -0,0 +1,318 @@
+}
+```
+
+Now this choice means that the SQL required to execute this query is now spread over two places in
````
```suggestion
Now, this choice means that the SQL required to execute this query is now spread over two places in
```

haletothewood commented on src/_posts/2020-01-14-fear-driven-development.md:
````diff
@@ -0,0 +1,318 @@
+}
+```
+
+Fortunately your test might catch that extra `0` you added and you can revert your change. No fear.
````
```suggestion
Fortunately, your test might catch that extra `0` you added and you can revert your change. No fear.
```

haletothewood reviewed (commented):
I ran it through grammarly. I liked the post a lot! Nice work 👍

Commit:
Grammar changes.

Commit:
Change date and time of post.

Commit:
Rename file.

upsd commented:
Thank you for the feedback @smeny-codurance, @LauLlobet and @haletothewood 👍 

I appreciate you all taking the time to look at it 😃
